Bio/File.py

```diff
@@ -95,7 +95,8 @@ def _open_for_random_access(filename):
             # If it is BGZF, we support that
             return bgzf.BgzfReader(mode="rb", fileobj=handle)
         except ValueError as e:
-            assert "BGZF" in str(e)
+            if "BGZF" not in str(e):
+                raise ValueError("Expected 'e' to contain 'BGZF'.")
```

Comment:
The message here should be something about unexpected exception, perhaps simplest to just ``raise e`` instead?